server/etcdserver/api/v3rpc/grpc.go

```diff
@@ -75,9 +75,11 @@ func Server(s *etcdserver.EtcdServer, tls *tls.Config, interceptor grpc.UnarySer
 	pb.RegisterLeaseServer(grpcServer, NewQuotaLeaseServer(s))
 	pb.RegisterClusterServer(grpcServer, NewClusterServer(s))
 	pb.RegisterAuthServer(grpcServer, NewAuthServer(s))
+	maintenanceServer := NewMaintenanceServer(s)
+	pb.RegisterMaintenanceServer(grpcServer, maintenanceServer)
 
 	hsrv := health.NewServer()
-	pb.RegisterMaintenanceServer(grpcServer, NewMaintenanceServer(s, NewHealthNotifier(hsrv, s.Logger())))
+	NewHealthNotifier(hsrv, s, maintenanceServer.DefragNotify())
```

Comment:
Can you reverse the dependency? It's MaintenanceServer that should take a Notify interface, not reverse. 

Having maintainenaceServer pushish events is a decent idea, however it doesn't work here. We don't plan to have multiple listeners and the current would not work with them.

Reply:
Done